front/lib/api/assistant/conversation.ts

```diff
@@ -1655,3 +1669,27 @@ async function* streamRunAgentEvents(
     }
   }
 }
+
+async function isMessagesLimitReached({
```

Comment:
I rewrote the query: 

```sql
EXPLAIN ANALYZE
SELECT
	count("message"."id") AS "count"
FROM
	"messages" AS "message"
	INNER JOIN "conversations" AS "conversation" ON "message"."conversationId" = "conversation"."id"
		AND "conversation"."workspaceId" = 1
WHERE
	"message"."agentMessageId" IS NOT NULL;
```
Aggregate  (cost=24.39..24.40 rows=1 width=8) (actual time=0.399..0.401 rows=1 loops=1)
  ->  Hash Join  (cost=5.16..23.50 rows=357 width=4) (actual time=0.099..0.363 rows=361 loops=1)
"        Hash Cond: (message.""conversationId"" = conversation.id)"
        ->  Seq Scan on messages message  (cost=0.00..17.36 rows=360 width=8) (actual time=0.023..0.197 rows=362 loops=1)
"              Filter: (""agentMessageId"" IS NOT NULL)"
              Rows Removed by Filter: 379
        ->  Hash  (cost=3.59..3.59 rows=126 width=4) (actual time=0.068..0.068 rows=127 loops=1)
              Buckets: 1024  Batches: 1  Memory Usage: 13kB
              ->  Seq Scan on conversations conversation  (cost=0.00..3.59 rows=126 width=4) (actual time=0.007..0.042 rows=127 loops=1)
"                    Filter: (""workspaceId"" = 1)"
                    Rows Removed by Filter: 1
Planning Time: 2.756 ms
Execution Time: 0.443 ms
```

```

Reply:
The seq scan on workspaceId = 1 looks fishy? there is no index here?

Can you try to explain the previous query to compare :grimacing: 

Also is this explain coming from production? (you can run them on metabase) Probably worth running it on alan's workspace just to see?

Reply:
Nope this is from my local 😬

Reply:
I would run the query on recent workspaces that have conversations, such as the one I did above, e.g. workspace 10355

Reply:
It might be indeed the case that the query you wrote was more efficient :+1: Would probably have deserved a comment  since we both quasi independently didn't understand the query while reviewing :+1:

Reply:
alan has lots of convos so the count is longer; the count here is restricted to < 100 messages so workspaces with ~ hundreds of messages are the good test

Reply:
> It might be indeed the case that the query you wrote was more efficient 👍 Would probably have deserved a comment since we both quasi independently didn't understand the query while reviewing 👍

Ack; have to leave now but will add the com when I get time :+1:

Reply:
Query to count messages of workspace, on LANA's workspace

Version with [count on conversation](https://metabase.dust.tt/question#eyJkYXRhc2V0X3F1ZXJ5Ijp7InR5cGUiOiJuYXRpdmUiLCJuYXRpdmUiOnsicXVlcnkiOiJFWFBMQUlOIEFOQUxZWkVcblNFTEVDVCBjb3VudChcImNvbnZlcnNhdGlvblwiLlwiaWRcIikgQVMgXCJjb3VudFwiIFxuRlJPTSBcImNvbnZlcnNhdGlvbnNcIiBBUyBcImNvbnZlcnNhdGlvblwiIFxuSU5ORVIgSk9JTiBcIm1lc3NhZ2VzXCIgQVMgXCJtZXNzYWdlc1wiIE9OIFwiY29udmVyc2F0aW9uXCIuXCJpZFwiID0gXCJtZXNzYWdlc1wiLlwiY29udmVyc2F0aW9uSWRcIiBcbklOTkVSIEpPSU4gXCJhZ2VudF9tZXNzYWdlc1wiIEFTIFwibWVzc2FnZXMtPmFnZW50TWVzc2FnZVwiIE9OIFwibWVzc2FnZXNcIi5cImFnZW50TWVzc2FnZUlkXCIgPSBcIm1lc3NhZ2VzLT5hZ2VudE1lc3NhZ2VcIi5cImlkXCIgQU5EIFwibWVzc2FnZXMtPmFnZW50TWVzc2FnZVwiLlwic3RhdHVzXCIgPSAnc3VjY2VlZGVkJyBcbldIRVJFIFwiY29udmVyc2F0aW9uXCIuXCJ3b3Jrc3BhY2VJZFwiID0gNDUyMDsiLCJ0ZW1wbGF0ZS10YWdzIjp7fX0sImRhdGFiYXNlIjo0fSwiZGlzcGxheSI6InNjYWxhciIsInBhcmFtZXRlcnMiOltdLCJ2aXN1YWxpemF0aW9uX3NldHRpbmdzIjp7fX0=).
New version with [count on messages](https://metabase.dust.tt/question#eyJkYXRhc2V0X3F1ZXJ5Ijp7InR5cGUiOiJuYXRpdmUiLCJuYXRpdmUiOnsicXVlcnkiOiJFWFBMQUlOIEFOQUxZWkVcblNFTEVDVCBjb3VudChcIm1lc3NhZ2VcIi5cImlkXCIpIEFTIFwiY291bnRcIlxuRlJPTVwibWVzc2FnZXNcIiBBUyBcIm1lc3NhZ2VcIlxuSU5ORVIgSk9JTiBcImNvbnZlcnNhdGlvbnNcIiBBUyBcImNvbnZlcnNhdGlvblwiIE9OIFwibWVzc2FnZVwiLlwiY29udmVyc2F0aW9uSWRcIiA9IFwiY29udmVyc2F0aW9uXCIuXCJpZFwiIEFORCBcImNvbnZlcnNhdGlvblwiLlwid29ya3NwYWNlSWRcIiA9IDQ1MjAgLS0gQUxhblxuV0hFUkVcblx0XCJtZXNzYWdlXCIuXCJhZ2VudE1lc3NhZ2VJZFwiIElTIE5PVCBOVUxMOyIsInRlbXBsYXRlLXRhZ3MiOnt9fSwiZGF0YWJhc2UiOjR9LCJkaXNwbGF5IjoidGFibGUiLCJwYXJhbWV0ZXJzIjpbXSwidmlzdWFsaXphdGlvbl9zZXR0aW5ncyI6e319).